src/main/java/com/faforever/client/reporting/AutoReportedException.java

```diff
@@ -0,0 +1,13 @@
+package com.faforever.client.reporting;
+
+import io.sentry.Sentry;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+
+public class AutoReportedException extends RuntimeException {
+    public AutoReportedException(Throwable cause) {
+      Sentry.capture(cause);
```

Comment:
Constructors should not call external code, especially code that does network requests.

The proper way to handle uncaught exceptions is to configure `Thread.setDefaultUncaughtExceptionHandler` (since there will also be other uncaught exceptions other than the ones created by NoCatch). Also, see `AsyncConfig` where a `SimpleAsyncUncaughtExceptionHandler` is configured.

Reply:
Sentry already implements the interface
https://github.com/getsentry/sentry-java/blob/38aaffa2d863fdc858cc5f9b24ff2e87d14b5a7a/sentry/src/main/java/io/sentry/SentryUncaughtExceptionHandler.java#L13
I do agree that the constructor shouldn't be calling the error, this was more of a way for me to do end to end testing